packages/backend/src/apps/m365-excel/actions/get-cell-values/index.ts

```diff
@@ -0,0 +1,125 @@
+import type { IRawAction } from '@plumber/types'
+
+import { generateStepError } from '@/helpers/generate-step-error'
+
+import WorkbookSession from '../../common/workbook-session'
+
+import type { DataOut } from './data-out'
+import getDataOutMetadata from './get-data-out-metadata'
+
+const action: IRawAction = {
+  name: 'Get cell value',
+  key: 'get-cell-values',
+  description: 'Gets the value of cells in a spreadsheet',
+  arguments: [
+    {
+      key: 'fileId',
+      label: 'Excel File',
+      required: true,
+      description: 'File to edit',
+      type: 'dropdown' as const,
+      variables: false,
+      source: {
+        type: 'query' as const,
+        name: 'getDynamicData' as const,
+        arguments: [
+          {
+            name: 'key',
+            value: 'listFiles',
+          },
+        ],
+      },
+    },
+    {
+      key: 'worksheetId',
+      label: 'Worksheet',
+      required: true,
+      description: 'Worksheet to query',
+      type: 'dropdown' as const,
+      variables: false,
+      source: {
+        type: 'query' as const,
+        name: 'getDynamicData' as const,
+        arguments: [
+          {
+            name: 'key',
+            value: 'listWorksheets',
+          },
+          {
+            name: 'parameters.fileId',
+            value: '{parameters.fileId}',
+          },
+        ],
+      },
+    },
+    {
+      label: 'Values',
+      key: 'cells',
+      type: 'multirow' as const,
+      required: true,
+      // We need to make 1 separate request for each cell, so limit to 3 as a
+      // balance between convenience and API quota usage.
+      description: 'Specify cells you want to get the values of. (Max 3)',
+      subFields: [
+        {
+          label: 'Cell Address (e.g. A1, B10)',
```

Comment:
nit: the e.g. made me think i could comma separate the cell addresses. i think (e.g. A1) is clear enough

Reply:
agreed, done